src/main/java/com/aws/greengrass/deployment/DeploymentConfigMerger.java

```diff
@@ -147,6 +156,21 @@ private void updateActionForDeployment(Map<String, Object> newConfig, Deployment
             return;
         }
 
+        // As long as the timeout is not 0, we will try to run the connectivity validation
+        DeploymentConfigurationValidationPolicy policy = deployment.getDeploymentDocumentObj()
+                .getConfigurationValidationPolicy();
+        if (policy != null && Coerce.toInt(policy.timeoutInSeconds()) != 0) {
+            try (Context context = new Context()) {
+                ConnectivityValidator connectivityValidator = new ConnectivityValidator(kernel, context, mqttClient,
+                        factory, securityService, newConfig, deployment.getDeploymentDocumentObj().getTimestamp());
+                if (!connectivityValidator.validate(totallyCompleteFuture, deviceConfiguration)) {
+                    return;
+                }
+            } catch (Throwable e) {
+                logger.atInfo().cause(e).log("Unexpected exception during connectivity validation");
+            }
+        }
+
```

Comment:
check for interruption here before continuing. `Thread.currentThread().interrupted()`